pilot/pkg/networking/ambientgen/ztunnelgen.go

```diff
@@ -1007,14 +979,14 @@ func outboundPodTunnelCluster(proxy *model.Proxy, push *model.PushContext, sa st
 		TransportSocket: &core.TransportSocket{
 			Name: "envoy.transport_sockets.tls",
 			ConfigType: &core.TransportSocket_TypedConfig{TypedConfig: protoconv.MessageToAny(&tls.UpstreamTlsContext{
-				CommonTlsContext: buildCommonTLSContext(proxy, workload, push, false),
+				CommonTlsContext: buildCommonTLSContext(proxy, identityOverride, push, false),
```

Comment:
When will it be overridden?

Reply:
This is just replacing `workload` which we extract spiffe+podname+podID to just a plain `spiffe`. Naming may be confusing... its overriding `proxy` (ztunnel)'s though